File: lib/data/models/dashboard_kpi.dart
// A KPI tile shown on the home screen. The visible KPIs vary by role.

enum KpiStatus {
  ok,
  warning,
  alert,
}

enum KpiTrend {
  up,
  down,
  flat,
}

class DashboardKpi {
  final String id;
  final String label;
  final String value;
  final String? unit;
  final KpiStatus status;
  final KpiTrend? trend;
  // Short helper text shown below the value (e.g., the comparison window).
  final String? helper;

  const DashboardKpi({
    required this.id,
    required this.label,
    required this.value,
    required this.status,
    this.unit,
    this.trend,
    this.helper,
  });
}

File: Data/Models/DashboardKpi.cs
namespace OrquestadorFrontend.Data.Models;

public enum KpiStatus
{
    Ok,
    Warning,
    Alert
}

public enum KpiTrend
{
    Up,
    Down,
    Flat
}

// A KPI tile shown on the home screen. The visible KPIs vary by role.
public sealed record DashboardKpi(
    string Id,
    string Label,
    string Value,
    KpiStatus Status,
    string? Unit = null,
    KpiTrend? Trend = null,
    string? Helper = null,
    // 0.0–1.0 progress toward target; null = no progress bar shown.
    double? Progress = null);
